src/sagemaker/estimator.py

```diff
@@ -521,7 +521,9 @@ def __init__(self, entry_point, source_dir=None, hyperparameters=None, enable_cl
         super(Framework, self).__init__(**kwargs)
         self.source_dir = source_dir
         self.entry_point = entry_point
-        self.enable_cloudwatch_metrics = enable_cloudwatch_metrics
+        if enable_cloudwatch_metrics:
+            logging.warn('enable_cloudwatch_metrics is now deprecated and will be removed in the future.')
```

Comment:
Could you use `warnings.warn` instead, with a `DeprecationWarning`?

https://docs.python.org/3/library/warnings.html#warnings.warn